### DISCUSSION TITLE
Should we remove unused visual script from code base?

### DISCUSSION THREAD

**Discussion by Doprez** [stride3d/stride#1611] [Ideas]
I have been looking aroung the Stride source code to familiarize myself with it a bit better and I see some code that is unused. The best example is the Visual script references from the VisualScriptAsset class, Manio has confirmed to me that this was an attempt at starting visual scripting pre-open ssource.

Can we remove references to this kind of code in a PR especially since I think vvvv seems like it may be a good alternative that currently exists and works well.

I will go ahead and make the PR if this makes sense, I know the current stride team isnt full time and are volunteers so stuff like this can easily be left behind, but I think getting rid of useless code could be a benefit to readability.

![image](https://user-images.githubusercontent.com/73259914/219065853-567aa57f-26b0-4d6a-878c-b2c0ece1315e.png)

**Comment by NicusorN5** [comment]
Why not start writing a visual scripting system from there?

**Reply by Doprez** [reply]
I guess my thought process is simplify the current stuff that exists but isnt used to give less hassle to potential contributors and less things to worry about. 
maybe even instead of removing the code, maybe you're right and a simple comment description could be just as good. 

Im just trying to think of ways to make the process easier on potential contributers and make the code more appealing rather than bloated looking.

**Comment by atcarter714** [comment]
You both have good points ... if we don't need it, then it should go, but if someone actually wants to build it then it should probably stay and get refactored and extended. If nobody can build it anytime soon then it may be best to take it out the active branch and restore it later if it's actually needed.

**Comment by manio143** [comment]
Yup, we're using git, so nothing should be lost once removed. I was actually curious myself about that existing code which seems to be integrating with roslyn through an API used otherwise for scripts editor and project watching for reloading assemblies (see `RoslynHost` and `RoslynWorkspace` usage). There may be parts of it that are in good shape and can be reused, but at the moment I think removing it is the correct move.

**Reply by atcarter714** [reply]
> Yup, we're using git, so nothing should be lost once removed. I was actually curious myself about that existing code which seems to be integrating with roslyn through an API used otherwise for scripts editor and project watching for reloading assemblies (see `RoslynHost` and `RoslynWorkspace` usage). There may be parts of it that are in good shape and can be reused, but at the moment I think removing it is the correct move.


On second thought ... I'm new to Stride and its source, but you could say I'm rather capable lol. Does Stride have any kind of simple node tool for making stuff like little animation state controller graphs and such? I haven't seen one yet at least ... that might be of some interest to me sometime soon

**Reply by manio143** [reply]
There's no animation editor currently. The closest UI tooling for graph based workflows is the implementation of GraphicsCompositor editor which visualizes the render dependency graph. It could potentially be reused for other means.

**Comment by Doprez** [comment]
I also decided to see if anyone has even made visual scripting for C# and I did find this neat tool https://github.com/RobinKa/netprints but it seems long dead for the moment.
